src/service/extensions-impl.js

```diff
@@ -168,12 +175,17 @@ export class Extensions {
   /**
    * Waits for the previously included extension to complete
    * loading/registration.
+   * @param {!Window} win
    * @param {string} extensionId
-   * @return {!Promise<!ExtensionDef>}
+   * @param {number=} opt_timeout
+   * @return {!Promise<?ExtensionDef>}
    */
-  waitForExtension(extensionId) {
-    return this.waitFor_(this.getExtensionHolder_(
-        extensionId, /* auto */ false));
+  waitForExtension(win, extensionId, opt_timeout) {
+    return /** @type {!Promise<?ExtensionDef>} */ (
+      Services.timerFor(win).timeoutPromise(opt_timeout || LOAD_TIMEOUT,
```

Comment:
do we ever want a case where it doesn't time out?

Reply:
Maybe the default should be 8 seconds.

The 3 seconds we use when it is in the critical path for users.

Reply:
geo may be critical path because consent depends on it.  Do i need to ad service specific timeout because by definition the geo extension can't set it?

The other way to do this would be pick a longish timeout and put the onus on the caller to do their own timeout in their code if they want it shorter.